crypto/rand/rand_lib.c

```diff
@@ -240,14 +240,22 @@ int RAND_pseudo_bytes(unsigned char *buf, int num)
 int RAND_status(void)
 {
     EVP_RAND_CTX *rand;
+    int state;
     const RAND_METHOD *meth = RAND_get_rand_method();
 
     if (meth != NULL && meth != RAND_OpenSSL())
         return meth->status != NULL ? meth->status() : 0;
 
     if ((rand = RAND_get0_primary(NULL)) == NULL)
-        return EVP_RAND_STATE_UNINITIALISED;
-    return EVP_RAND_state(rand) == EVP_RAND_STATE_READY;
+        return 0;
+    state = EVP_RAND_state(rand);
+    if (state == EVP_RAND_STATE_UNINITIALISED) {
+        if (!EVP_RAND_instantiate(rand, 0, 0, NULL, 0)) {
+            ;   /* Nothing to do here */
+        }
+        state = EVP_RAND_state(rand);
+    }
```

Comment:
Hmmm... actually my understanding was that RAND_status() should _not_ attempt an instantiate of an uninstantiated random generator, but only check the state. Instead, an automatic instantiation should happen on first use of `RAND_get0_primary()`, as it used to happen in `RAND_DRBG_get0_master()`, see my https://github.com/openssl/openssl/issues/12714#issuecomment-680283913:

> > RAND_status() must trigger implicit initialization the same way as RAND_bytes() does.
> 
> I don't recall that RAND_status() ever triggered an automatic instantiation and I don't think it should do so. The function is only meant to query the RNG's state. What actually is broken is the automatic instantiation of the primary (formerly: master) DRBG on startup. In prerelease alpha6, it is triggered on the first call to `RAND_DRBG_get0_master()` and carried out by `drbg_setup()`:

Reply:
Just imagine that someone would call `EVP_RAND_state(RAND_get0_primary(NULL))` directly instead of RAND_status() and find out that the state is EVP_RAND_STATE_UNINITIALISED. That would be quite surprising, wouldn't it?

Reply:
I misunderstood the intent.  Fix up pushed.